src/domain/multipart-middleware.js

```diff
@@ -0,0 +1,64 @@
+import mime from 'mime';
+import multer from 'multer';
+import multerS3 from 'multer-s3';
+import { Router } from 'express';
+import httpErrors from 'http-errors';
+import prettyBytes from 'pretty-bytes';
+import permissions, { hasUserPermission } from './permissions.js';
+import { STORAGE_FILE_UPLOAD_LIMIT_IN_BYTES } from './constants.js';
+import { createUniqueStorageFileName, getTemporaryUploadsPath } from '../utils/storage-utils.js';
+
+const { BadRequest } = httpErrors;
+
+const SIZE_RESERVED_FOR_REMAINING_CONTENT_IN_BYTES = 500 * 1000;
+const EFFECTIVE_REQUEST_LIMIT_IN_BYTES = STORAGE_FILE_UPLOAD_LIMIT_IN_BYTES + SIZE_RESERVED_FOR_REMAINING_CONTENT_IN_BYTES;
+
+function createMultipartMiddleware({ cdn, fileField, bodyField = null, multipleFiles = false, allowUnlimitedUploadForElevatedRoles = false }) {
+  const checkUploadLimitMiddleware = (req, _res, next) => {
+    const requestSize = !!req.headers['content-length'] && Number(req.headers['content-length']);
+    const allowUnrestricted = allowUnlimitedUploadForElevatedRoles && hasUserPermission(req.user, permissions.UPLOAD_WITHOUT_SIZE_RESTRICTION);
+    return requestSize && requestSize > EFFECTIVE_REQUEST_LIMIT_IN_BYTES && !allowUnrestricted
+      ? next(new BadRequest(`Upload limit exceeded: limit ${prettyBytes(EFFECTIVE_REQUEST_LIMIT_IN_BYTES)}, requested ${prettyBytes(requestSize)}.`))
+      : next();
+  };
+
+  const multipartParser = multer({
+    storage: multerS3({
+      s3: cdn.s3Client.client,
+      bucket: cdn.bucketName,
+      key: (_req, file, cb) => cb(null, `${getTemporaryUploadsPath()}/upload_${createUniqueStorageFileName(file.originalname)}`),
```

Comment:
Is the `upload_` prefix for the file name necessary for AWS' TTL?

Reply:
Yes, that was the intention, in order to have the `__DIRMARKER__` excluded from automatic deletion.

Reply:
Got it 👍 Was wondering why the path/"folder" name was not enough.